### ISSUE TITLE
[Package Request] - PHP 8

### ISSUE THREAD

**Issue by stewartsmith** [amazonlinux/amazon-linux-2022#11]
**What package is missing from Amazon Linux 2022? Please describe and include package name.**
PHP 8.0.

**Is this an update to existing package or new package request?**
Addition of PHP 8.0 alongside PHP 7.4, both to have a support period matching upstream represented in supportinfo.xml

**Is this package available in Amazon Linux 2?**
Yes, PHP 8 is available as an AL2 Extra.

**Comment by joelbarnard** [comment]
As a FYI for consideration, PHP 8.1 has recently be released https://www.php.net/releases/8.1/en.php

**Comment by skyzyx** [comment]
PHP versions don’t receive support from the upstream vendor for 5 years — it's closer to 3 years. _Maybe_ as an extra it'd make sense?

Honestly, I think https://hub.docker.com/_/php is a far-better solution — both for the package maintainers as well as the PHP community.

**Comment by joelbarnard** [comment]
Just wanted to follow up on this, and see if there was any plans to include it.
Seems odd that PHP 7.4 (which went EOL before your last post) would be included, but 8 wouldn't.

The FAQs even call out PHP specifically as something that would get quarterly updates.
https://aws.amazon.com/linux/amazon-linux-2022/faqs/
Q: What is included in major and minor releases?
A:  ... "Quarterly minor releases will include security updates, bug fixes, and new features and packages. Examples of changes in the minor releases **include latest language runtimes, like PHP** and other popular software packages such as Ansible and Docker." ...

**Comment by skyzyx** [comment]
PHP 7.4 is supported through the end of 2022.

PHP 7.3 went EOL at the end of 2021.

And despite the thumbs-down votes on my previous comment, I still stand by what I said. It's time for a grown-up server configuration.

**Comment by srbt** [comment]
[I asked](https://forum.remirepo.net/viewtopic.php?pid=12197) a person who maintains packages of different versions of php for different versions of Fedora if he could support Amazon Linux 2022. He had [some critical remarks](https://forum.remirepo.net/viewtopic.php?pid=12194#p12194) about the combination of php and openssl versions and refused to participate. For our project I use images that combine php, nodejs, python and c++. With amazonlinux: 2 I did well, but it's really a bit outdated.

Now I'm struggling to compile php 8.0 - passed, but `make test` reports 95 failures.

I'm using Amazon Linux, because I want migrate to aws and I hope, that Amazon Linux better fit to aws environment.

Will you recommend me to use another distribution too?

**Comment by RichardDale** [comment]
Al2022 release ver 2022.0.20220628 has php8 - but the package has no contents, unfortunately, so no PHP is functional on that version at all. :-(

**Comment by stewartsmith** [comment]
If you're looking for mod_php, that's no longer available. But the `php-fpm` package is.

There appears to be a bug in our repo metadata where we're not pulling in the `Recommends` that exists in the `php` RPM that would have auto installed the `php-fpm` package. I'll chase that up after the weekend.

**Comment by RichardDale** [comment]
I mean that the php-8.1.7-1.amzn2022.0.4.x86_64.rpm basically has no contents:

```
# dnf install php
Last metadata expiration check: 5:38:05 ago on Sun 03 Jul 2022 10:39:22 PM UTC.
Dependencies resolved.
======================================================================================================================
 Package                   Architecture          Version                             Repository                  Size
======================================================================================================================
Installing:
 php                       x86_64                8.1.7-1.amzn2022.0.4                amazonlinux                 17 k
Installing dependencies:
 php-common                x86_64                8.1.7-1.amzn2022.0.4                amazonlinux                673 k

# rpm -q -l php-8.1.7-1.amzn2022.0.4.x86_64
(contains no files)
```

I was expecting a php executable somewhere!

(php development isn't really my thing, so perhaps things have changed in the typical PHP environment in the last few years)...

**Comment by srbt** [comment]
For now I solved this with compilation from sources. With mod_php.
[Dockerfile.txt](https://github.com/amazonlinux/amazon-linux-2022/files/9037731/Dockerfile.txt)
[openssl.c.patch.txt](https://github.com/amazonlinux/amazon-linux-2022/files/9037734/openssl.c.patch.txt)
 (IMAGE_AMLI is amazonlinux:2022, VER_PHP is in my case 8.0.19)

**Comment by stewartsmith** [comment]
The repo metadata problem has been fixed in the `2022.0.20220728` release.

In a container:
```
bash-5.1# dnf install php
Last metadata expiration check: 0:03:43 ago on Tue Aug  2 00:49:48 2022.
Dependencies resolved.
================================================================================
 Package              Arch    Version                        Repository    Size
================================================================================
Installing:
 php8.1               x86_64  8.1.7-1.amzn2022.0.1           amazonlinux   16 k
Installing dependencies:
 apr                  x86_64  1.7.0-9.amzn2022               amazonlinux  121 k
 apr-util             x86_64  1.6.1-16.amzn2022.0.1          amazonlinux   97 k
 cracklib             x86_64  2.9.6-27.amzn2022              amazonlinux   84 k
 dbus                 x86_64  1:1.12.20-3.amzn2022           amazonlinux  9.3 k
 dbus-broker          x86_64  29-2.amzn2022                  amazonlinux  175 k
 dbus-common          noarch  1:1.12.20-3.amzn2022           amazonlinux   16 k
 device-mapper        x86_64  1.02.185-1.amzn2022.0.1        amazonlinux  141 k
 device-mapper-libs   x86_64  1.02.185-1.amzn2022.0.1        amazonlinux  181 k
 generic-logos-httpd  noarch  18.0.0-12.amzn2022             amazonlinux   19 k
 gzip                 x86_64  1.10-5.amzn2022                amazonlinux  152 k
 httpd-core           x86_64  2.4.54-3.amzn2022.0.1          amazonlinux  1.4 M
 httpd-filesystem     noarch  2.4.54-3.amzn2022.0.1          amazonlinux   14 k
 httpd-tools          x86_64  2.4.54-3.amzn2022.0.1          amazonlinux   82 k
 kmod-libs            x86_64  29-2.amzn2022.0.2              amazonlinux   62 k
 libargon2            x86_64  20171227-9.amzn2022            amazonlinux   29 k
 libbrotli            x86_64  1.0.9-4.amzn2022               amazonlinux  325 k
 libdb                x86_64  5.3.28-49.amzn2022             amazonlinux  754 k
 libeconf             x86_64  0.4.0-1.amzn2022               amazonlinux   28 k
 libedit              x86_64  3.1-38.20210714cvs.amzn2022    amazonlinux  108 k
 libfdisk             x86_64  2.37.4-1.amzn2022              amazonlinux  156 k
 libpwquality         x86_64  1.4.4-6.amzn2022               amazonlinux  106 k
 libseccomp           x86_64  2.5.3-1.amzn2022               amazonlinux   72 k
 libsemanage          x86_64  3.2-1.amzn2022                 amazonlinux  121 k
 libutempter          x86_64  1.2.1-4.amzn2022               amazonlinux   27 k
 libxslt              x86_64  1.1.34-5.amzn2022              amazonlinux  246 k
 mailcap              noarch  2.1.49-3.amzn2022              amazonlinux   34 k
 nginx-filesystem     noarch  1:1.22.0-1.amzn2022.0.1        amazonlinux   12 k
 oniguruma            x86_64  6.9.7.1-1.amzn2022             amazonlinux  231 k
 pam                  x86_64  1.5.1-8.amzn2022.0.1           amazonlinux  543 k
 php8.1-common        x86_64  8.1.7-1.amzn2022.0.1           amazonlinux  673 k
 shadow-utils         x86_64  2:4.8.1-9.amzn2022             amazonlinux  1.1 M
 systemd              x86_64  250.7-1.amzn2022.0.2           amazonlinux  4.2 M
 systemd-libs         x86_64  250.7-1.amzn2022.0.2           amazonlinux  617 k
 systemd-pam          x86_64  250.7-1.amzn2022.0.2           amazonlinux  337 k
 util-linux           x86_64  2.37.4-1.amzn2022              amazonlinux  2.2 M
 util-linux-core      x86_64  2.37.4-1.amzn2022              amazonlinux  435 k
 xkeyboard-config     noarch  2.33-1.amzn2022                amazonlinux  784 k
Installing weak dependencies:
 apr-util-bdb         x86_64  1.6.1-16.amzn2022.0.1          amazonlinux   14 k
 apr-util-openssl     x86_64  1.6.1-16.amzn2022.0.1          amazonlinux   16 k
 cryptsetup-libs      x86_64  2.4.3-2.amzn2022               amazonlinux  433 k
 diffutils            x86_64  3.8-1.amzn2022                 amazonlinux  406 k
 httpd                x86_64  2.4.54-3.amzn2022.0.1          amazonlinux   48 k
 libbpf               x86_64  2:0.3.0-1.amzn2022             amazonlinux  116 k
 libxkbcommon         x86_64  1.3.0-1.amzn2022               amazonlinux  147 k
 mod_http2            x86_64  1.15.24-1.amzn2022.0.1         amazonlinux  152 k
 mod_lua              x86_64  2.4.54-3.amzn2022.0.1          amazonlinux   62 k
 php8.1-cli           x86_64  8.1.7-1.amzn2022.0.1           amazonlinux  3.5 M
 php8.1-fpm           x86_64  8.1.7-1.amzn2022.0.1           amazonlinux  1.8 M
 php8.1-mbstring      x86_64  8.1.7-1.amzn2022.0.1           amazonlinux  480 k
 php8.1-opcache       x86_64  8.1.7-1.amzn2022.0.1           amazonlinux  382 k
 php8.1-pdo           x86_64  8.1.7-1.amzn2022.0.1           amazonlinux   91 k
 php8.1-xml           x86_64  8.1.7-1.amzn2022.0.1           amazonlinux  148 k
 qrencode-libs        x86_64  4.1.1-2.amzn2022.0.1           amazonlinux   66 k
 systemd-networkd     x86_64  250.7-1.amzn2022.0.2           amazonlinux  562 k
 systemd-resolved     x86_64  250.7-1.amzn2022.0.2           amazonlinux  278 k

Transaction Summary
================================================================================
Install  56 Packages

Total download size: 24 M
Installed size: 95 M
Is this ok [y/N]: 
```
